include/bitcoin/blockchain/interface/safe_chain.hpp

```diff
@@ -52,6 +52,13 @@ class BCB_API safe_chain
     // Smart pointer parameters must not be passed by reference.
     typedef std::function<void(const code&, block_const_ptr, size_t)>
         block_fetch_handler;
+
+    typedef std::function<void(const code&, block_const_ptr, const hash_list&, uint64_t, size_t)>
```

Comment:
I would put it in the following order:
 block (ptr and height), hash_list, and block_size:

std::function<void(const code&, block_const_ptr, size_t, const hash_list&, uint64_t)>

to be consistent with the existing functions...